maps_dashboards/public/components/add_layer_panel/add_layer_panel.tsx

```diff
@@ -40,10 +40,13 @@ export const AddLayerPanel = ({
 }: Props) => {
   const [isAddNewLayerModalVisible, setIsAddNewLayerModalVisible] = useState(false);
   const [highlightItem, setHighlightItem] = useState<Layer | null>(null);
+  const [nextLayerIndex, setNextLayerIndex] = useState<number>(1);
 
   function onClickAddNewLayer(layerType: string) {
     const initLayerConfig = getLayerConfigMap()[layerType];
+    initLayerConfig.name = 'New layer ' + nextLayerIndex;
```

Comment:
How about use `layers.length + 1` for the new layer number? The nextLayerIndex will keep increasing even if I create a new layer but discard it.

Reply:
Sure. But i don't think we have access to total layer inside Add layer panel, if not, how to get layer count without adding new dependency?

Reply:
You could pass layers into AddLayerPanel component Props and use it

Reply:
I got error message from `initLayerConfig.name = 'New layer ' + nextLayerIndex;`, we might need to change not use enum for DASHBOARDS_MAPS_LAYER_NAME
`
TS2322: Type 'string' is not assignable to type 'DASHBOARDS_MAPS_LAYER_NAME'.
`

Reply:
kept default value as empty string.